src/server/Monte.AuthServer/Services/UserService.cs

```diff
@@ -177,12 +181,44 @@ public IdentityErrorException(IReadOnlyDictionary<string, string[]> errors)
     }
 }
 
+public class Result
+{
+    public string? ErrorMessage { get; set; }
+    public ErrorType ErrType { get; set; }
+
+    public Result(string? Error, ErrorType type)
+    {
+        ErrorMessage = Error;
+        ErrType = type;
+    }
+
+    public Result()
+    {
+        ErrorMessage = string.Empty;
+        ErrType = ErrorType.None;
+    }
+
+    public enum ErrorType
+    {
+        None, NotFound, BadRequest
+    }
+}
+
+public class Result<T> : Result
+{
+    public T? Object { get; set; }
+
+    public Result(T? Obejct, string? Error = null) : base(Error, ErrorType.None)
```

Comment:
typo, also it is uncommon to start method and constructor parameter names with a capital letter

Reply:
also you can create an object with an error message but without an error type which seems like an error in design